src/faultlog/faultlog_main.cpp

```diff
@@ -373,14 +357,31 @@ int main(int argc, char** argv)
                     if (isHostProgressStateRunning(bus))
                     {
                         lg2::info("faultlog poweron timer host reached running "
-                                  "state. create fautlog pel ");
-                        createHostPowerOnNagPel(bus);
+                                  "state consider it a bmcreboot");
+                        createNagPel(
+                            bus, unresolvedRecords,
+                            !IGNORE_PWR_FAN_PEL); // add as it is bmcreboot
                         timer.setEnabled(false);
                         exit(EXIT_SUCCESS);
                     }
                 };
                 Timer timer(event, std::move(timerCb), hostStateCheckTimeout);
                 timer.setEnabled(true);
+
+                // wait for host to reach runtime
+                lg2::info("faultlog host is not in running state create watch "
+                          "for progress state");
+                std::unique_ptr<sdbusplus::bus::match_t> _hostStatePropWatch =
```

Comment:
Just a query (might be silly). Where this `_hostStatePropWatch` is been used?

Reply:
it will register a callback method called propertiesChanged which will be called when ever there is Boot.Progress change and as there i an event loop at line 386 it will not go out of scope.